### ISSUE TITLE
Timedomain error triggered by LIST type on Metabase

### ISSUE THREAD

**Issue by andrioni** [MaterializeInc/materialize#9374]
### What version of Materialize are you using?

From the shell:

```
materialize=> select mz_version();
       mz_version
-------------------------
 v0.10.1-dev (d68ab011e)
(1 row)
```
(#9366 as of some moment yesterday)

### How did you install Materialize?

* [x] Built from source

### What was the issue?

When trying to visualize a table or view which contains a column with the LIST type, the following error is returned:
```
Detail: The following relations in the query are outside the transaction's time domain: "pg_catalog.pg_namespace" "pg_catalog.pg_type" Only the following relations are available: "mz_catalog. [snipped]
```

### Is the issue reproducible? If so, please provide reproduction instructions.

Try to open `mz_catalog. mz_dataflow_operator_addresses` on Metabase.

### Please attach any applicable log files.

[Gist with pgwire=trace logs](https://gist.github.com/andrioni/eca100713ffa5ecd93e365519f89487b)

Select fragment of the queries being executed from the gist above:
```sql
BEGIN READ ONLY;

-- Metabase:: userID: 1 queryType: MBQL queryHash: 8095bed5fef1db0ed1f2a22cc64a98d10eb12e64155b6e70012947474d15a5fd
SELECT "mz_catalog"."mz_dataflow_operator_addresses"."id" AS "id", "mz_catalog"."mz_dataflow_operator_addresses"."worker" AS "worker", "mz_catalog"."mz_dataflow_operator_addresses"."address" AS "address" FROM "mz_catalog"."mz_dataflow_operator_addresses" LIMIT 2000;

SELECT n.nspname = ANY(current_schemas(true)), n.nspname, t.typname FROM pg_catalog.pg_type t JOIN pg_catalog.pg_namespace n ON t.typnamespace = n.oid WHERE t.oid = $1;
-- raw_params: [Some([0, 0, 64, 0])]

error code=25000 message=Transactions can only reference objects in the same timedomain. See https://materialize.com/docs/sql/begin/#same-timedomain-error
```

**Comment by andrioni** [comment]
I can still reproduce this issue on `main` (as of 0411718a4e99739c512f29ecb0999f20338508e6) with the queries above and on Metabase.

**Comment by mjibson** [comment]
This query is from: https://github.com/pgjdbc/pgjdbc/blob/db8257578336c7a5cc63ebc99fbdff44b7aac80c/pgjdbc/src/main/java/org/postgresql/jdbc/TypeInfoCache.java#L543

**Comment by morsapaes** [comment]
Was just re-testing the integration and double checked that both this and #9375 have been fixed in #10019. Closing the issue(s)!
